utils/python/CIME/build.py

```diff
@@ -94,8 +94,9 @@ def build_model(build_threaded, exeroot, clm_config_opts, incroot, complist,
 
     config_dir = os.path.join(cimeroot, "driver_cpl", "cime_config")
     f = open(file_build, "w")
+    bldroot = os.path.join(exeroot, "cpl", "obj")
     stat = run_cmd("%s/buildexe %s %s %s" %
-                   (config_dir, caseroot, bldroot, libroot),
+                   (config_dir, caseroot, libroot, bldroot),
```

Comment:
Again, this argument swapping looks suspicious, since buildexe has:

```python
    caseroot, bldroot, libroot = parse_input(sys.argv)
```

Again, if this is intentional, can you please add a comment describing why?